### PR TITLE
Markers for food scrap dropoff locations

### PR DESCRIPTION
**Pull request by LindseyWeiskopf**
Summary: I added markers onto my map for every food scrap dropoff location in NYC
[lweiskopf-step-2020.appspot.com](url)

In this iteration, I took the map that I created in my last PR and added the Maps API marker feature to it. I found a simple data set (food scrap dropoff locations in NYC) and cleaned the data such that it is a csv of coordinates. To integrate this data onto my map, I created an Arraylist in DropoffServlet.java that holds a list of ScrapDropoff objects. The objects (defined in ScrapDropoff.java) contain a latitude and longitude value representing the location of the dropoff. The servlet parses the csv file into ScrapDropoff objects that can be converted into JSON. 

The doGet() function in the servlet transforms the list of ScrapDropoff objects in JSON and sends that information to the /dropoff-data page to be fetched by script.js. 

Building off of the map initialized in the last iteration, I then place markers onto the map by fetching the JSON coordinates, and initializing a maps API marker object for each dropoff location by specifying the latitude and longitude attributes.

Finally, this map full of markers gets displayed on the index.html page upon load.

Please let me know if anything is unclear. Thanks so much!

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/ScrapDropoff.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.google.sps.data;
+
+// Represents the physical location of a Food scrap Dropoff in NYC
+public final class ScrapDropoff {
+
```
Comment (shaquille-hall) [inline portfolio/src/main/java/com/google/sps/data/ScrapDropoff.java:5]: nit: You can remove this extra line.

File: portfolio/src/main/java/com/google/sps/servlets/DropoffServlet.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+// Will send JSON formatted coordinates of dropoff locations to
```
Comment (shaquille-hall) [inline portfolio/src/main/java/com/google/sps/servlets/DropoffServlet.java:15]: nit: You can shorten this to "Sends" instead of "Will send"

File: portfolio/src/main/java/com/google/sps/servlets/DropoffServlet.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  // Create array list with parse csv values as ScrapDropoff objects
+  @Override
+  public void init(){
+    dropoffs = new ArrayList<>();
+    Scanner scanner = new Scanner(getServletContext().getResourceAsStream("/WEB-INF/dropoff-locations.csv"));
+    while (scanner.hasNextLine()) {
+      String line = scanner.nextLine();
+      String[] cells = line.split(",");
+
+      double lat = Double.parseDouble(cells[0]);
```
Comment (shaquille-hall) [inline portfolio/src/main/java/com/google/sps/servlets/DropoffServlet.java:31]: This looks like a good use case for @Autovalue, or a helper class  if you're interested in trying it out. It will help keep these array indexes named, and distinguishable. One obvious pitfall is mixing up the indices = setting lat = cell[1] and lon = cells[0]. With a helper object this can become lat = cells.lat etc.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -53,16 +53,27 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
-// Creates a map and adds it to the page.
+// Creates a map and adds it to the page with markers on compost-dropoff locations.
 function initMap() {
-  // Coordinates center map  around Baltimore, MD
-  var myLatlng = new google.maps.LatLng(39.412, -76,775);
+  // Coordinates centers map  around NYC
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:58]: As in dropoff centers?

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -53,16 +53,27 @@ function addRandomGreeting() {
...
+  var myLatlng = {lat: 40.763232, lng: -73.951047};
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:59]: The same as a previous comments (possibly in a different file), make these constants, with comments for why they're important.

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -53,16 +53,27 @@ function addRandomGreeting() {
...
     center: myLatlng, 
-    zoom: 12, 
-    mapTypeId: 'roadmap'
+    zoom: 12,
+    mapTypeControl: false 
   };
   const map = new google.maps.Map(document.getElementById('map'), mapOptions);
-}
-
+  var image = 'http://maps.google.com/mapfiles/kml/pal4/icon47.png';
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:66]: It's a good practice to define this as a constant, I imagine the're is no need for it to change at runtime.

File: portfolio/src/main/java/com/google/sps/data/ScrapDropoff.java (hunk 1 above)
Comment (LindseyWeiskopf) [reply portfolio/src/main/java/com/google/sps/data/ScrapDropoff.java]: Done.

File: portfolio/src/main/java/com/google/sps/servlets/DropoffServlet.java (hunk 3 above)
Comment (LindseyWeiskopf) [reply portfolio/src/main/java/com/google/sps/servlets/DropoffServlet.java]: I looked into autovalue, and to me it seemed like a helper class made more sense here because I can feed in the full line and parse in the constructor of the class. In what I read about autovalue, I wasn't sure how I could do that since the compiler creates so much of the functionality at runtime. I added a helper class implementation (Coordinate.java), but let me know if you think that autovalue would be better suited here and if there is a good place to start with that.

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (LindseyWeiskopf) [reply portfolio/src/main/webapp/script.js]: Done!

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (shaquille-hall) [reply portfolio/src/main/webapp/script.js]: Ahh. That is much clearer. Thank you!
